nexus/db-queries/src/db/datastore/rack.rs

```diff
@@ -793,60 +795,49 @@ impl DataStore {
     pub async fn nexus_external_addresses(
         &self,
         opctx: &OpContext,
+        blueprint: Option<&Blueprint>,
     ) -> Result<(Vec<IpAddr>, Vec<DnsZone>), Error> {
         opctx.authorize(authz::Action::Read, &authz::DNS_CONFIG).await?;
 
-        use crate::db::schema::external_ip::dsl as extip_dsl;
-        use crate::db::schema::service::dsl as service_dsl;
-
-        let err = OptionalError::new();
-        let conn = self.pool_connection_authorized(opctx).await?;
-        self.transaction_retry_wrapper("nexus_external_addresses")
-            .transaction(&conn, |conn| {
```

Comment:
I don't think it should matter?  The list of DNS zones cannot actually change today at all.  If it did, I think you'd already have a problem because we're constructing a list of DNS names based on those zones, and that could become out of date by the time we try to write the DNS changes.  The set of external IPs could, too, though only via Reconfigurator.  Eventually I think we want to have even the Silo DNS changes happen through Reconfigurator, and then we'll know it will at least converge to correct.  Anyway, since the list of DNS zones cannot change today, I think it makes sense to do this and revisit if/when we support that.